llama.h

```diff
@@ -468,6 +469,10 @@ extern "C" {
     // Print system information
     LLAMA_API const char * llama_print_system_info(void);
 
+    LLAMA_API void llama_dump_result_info_yaml(
+        FILE * stream, const llama_context * ctx, const char * output_str, const int * output_tokens,
+        int n_output_tokens, const float * probs, int n_probs);
+
```

Comment:
This should be just:

```suggestion
    LLAMA_API void llama_dump_context_yaml(FILE * stream, const llama_context * ctx);
```

The tools (`main`, `perplexity`, etc.) should be responsible to dump any specifics of their results, like `output_str`, `probs`, etc.

A secondary note is that passing `FILE *` through the API is not super great long term.
At some point, someone will need this info dumped in a buffer instead of file and if you have an API for dumping to a buffer, you no longer need the one for dumping to a file.
You can leave it like this since this is kind of dev-related functionality, but might consider changing it in the future if it sticks.

Reply:
I'm currently doing it like this because I want the properties to be alphabetically sorted per category. But if you remove the binary-specific properties you're essentially left with just the timings so maybe I'll just put them in a separate section.